### DISCUSSION TITLE
Simple Docker implementation

### DISCUSSION THREAD

**Discussion by JenningsAutomation** [webdriverio/webdriverio#11516] [Community]
Hello, I'm struggling to get a simple docker container off the ground. I would like to start with testing the default testcase that is there when you fire up webdriver from the getting started page. I need to implement a more complex solution at my job but for now I just want the see if the sample testcase can be built, ran and completed in docker. The sample Dockerfile in the documentation:

FROM ianwalter/puppeteer:latest
WORKDIR /app
ADD . /app

RUN npm install

CMD npx wdio

returns the following errors when I build it.:
ChromeDriver installation failed TypeError: fs.rmSync is not a function
    at copyIntoPlace (/app/node_modules/chromedriver/install.js:314:6)
    at install (/app/node_modules/chromedriver/install.js:73:11)
npm WARN my-new-project@ No description
npm WARN my-new-project@ No repository field.
npm WARN my-new-project@ No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! chromedriver@112.0.1 install: `node install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the chromedriver@112.0.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

i would like to build it from an official node image or ubuntu if possible. Any assistance would be appreciated. I am running Ubuntu 20.04 if it matters. I'd like to prove this out on chrome and firefox browsets

**Comment by erwinheitzman** [comment]
Have you tried using https://webdriver.io/docs/wdio-docker-service/ I think this is the easiest setup we have?

**Comment by JenningsAutomation** [comment]
That works, but it depends on npm and webdriver being installed locally. Then you run the selenium/standalone container and point your project to localhost:4444. I would like to have npm and webdriver running in a container.

**Reply by christian-bromann** [reply]
The `ianwalter/puppeteer` image seems outdated. I looked into creating a Docker image but was unsuccessful to have a setup for Chrome and Firefox. I was able to get Firefox running but not Chrome. I think it would be nice to offer WebdriverIO images if browser installed on them. Any contributions in that direction would be nice and appreciated.

**Reply by JenningsAutomation** [reply]
I would like to contribute with respect to docker. How can I get started?

**Reply by christian-bromann** [reply]
I think it would be great to start with a Dockerfile that is based on a common base e.g. ubuntu or alpine, that can setup as many browser as possible and would allow users to run their tests in it if they use it as base image. Next iteration would be to make variations of that, e.g. have images that only come with a single browser, that support different WebdriverIO versions or browser versions. One could look at https://github.com/cypress-io/cypress-docker-images for inspiration.
